CourseChamp/src/test/java/ca/mcgill/ecse428/CourseChamp/StepDefinitions/UserPortalStepDefinitions.java

```diff
@@ -0,0 +1,284 @@
+import ca.mcgill.ecse428.CourseChamp.model.Course;
+import ca.mcgill.ecse428.CourseChamp.model.CourseOffering;
+import ca.mcgill.ecse428.CourseChamp.model.Student;
+import ca.mcgill.ecse428.CourseChamp.model.Student.Major;
+import ca.mcgill.ecse428.CourseChamp.repository.AdminRepository;
+import ca.mcgill.ecse428.CourseChamp.repository.CourseOfferingRepository;
+import ca.mcgill.ecse428.CourseChamp.repository.CourseRepository;
+import ca.mcgill.ecse428.CourseChamp.repository.ReviewRepository;
+import ca.mcgill.ecse428.CourseChamp.repository.StudentRepository;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+
+import org.apache.catalina.connector.Response;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.ResponseEntity;
+
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class UserPortalStepDefinitions {
+    @Autowired
+    CourseRepository courseRepository;
+
+    @Autowired
+    CourseOfferingRepository courseOfferingRepository;
+
+    @Autowired
+    StudentRepository studentRepository;
+
+    @Autowired
+    ReviewRepository reviewRepository;
+
+    @Autowired
+    AdminRepository adminRepository;
+
+    @Autowired
+    StudentController studentController;
+
+    @Autowired
+    AdminController adminController;
+
+    @Autowired
+    private TestRestTemplate client;
+
+    private StudentRequestDto studentRequest;
+    private ResponseEntity<StudentResponseDto> studentResponse;
+
+    private AdminRequestDto adminRequest;
+    private ResponseEntity<AdminResponseDto> adminResponse;
+
+    private ResponseEntity<List> responseList;
+
+    private ResponseEntity<String> errorMessage;
+
+    
+    private ResponseEntity<ReviewResponseDto> reviewResponse;
+
+
+    @When("the student {string} attempts to change their username to {string}")
+    public void the_student_attempts_to_change_their_username(String string, String string2) {
+        // Write code here that turns the phrase above into concrete actions
+        ResponseEntity<StudentResponseDto> student = (ResponseEntity<StudentResponseDto>)studentController.getStudentByEmail("/student/"+string);
+        studentRequest = new StudentRequestDto();
+        studentRequest.setPassword(student.getBody().getPassword());
+        studentRequest.setMajor(student.getBody().getMajor());
+        studentRequest.setEmail(string);
+        studentRequest.setUsername(string2);
+
+        studentResponse = client.postForEntity("/student/update", studentRequest, StudentResponseDto.class);
```

Comment:
will need to add the endpoints "/student/update" and "/admin/update" during our sprint